self-hosting/env-variables.mdx

```diff
@@ -0,0 +1,80 @@
+---
+title: Configuration
+description: Complete list of environment variables for LangWatch self-hosting
+---
+
+# Environment Variables
+
+This page provides a comprehensive list of all environment variables that can be configured for LangWatch self-hosting deployments. These variables control various aspects of the application including authentication, database connections, external services, and feature flags.
+
+## Environment Variables Reference
+
+| Variable | Description | Required | Default |
+|----------|-------------|----------|---------|
+| **Core Configuration** |
+| `DATABASE_URL` | PostgreSQL database connection URL | Yes | - |
+| `NODE_ENV` | Application environment (development, test, production) | Yes | - |
+| `BASE_HOST` | Base URL of the application | Yes | - |
+| `API_TOKEN_JWT_SECRET` | Secret key for JWT token signing | Yes | - |
+| **Authentication (NextAuth)** |
+| `NEXTAUTH_PROVIDER` | Authentication provider (azure-ad, okta, auth0, etc.) | No | email |
+| `NEXTAUTH_SECRET` | Secret for NextAuth.js session encryption | Yes | - |
+| `NEXTAUTH_URL` | URL of the authentication service | Yes | - |
+| **Search & Storage** |
+| `ELASTICSEARCH_NODE_URL` | Elasticsearch cluster endpoint | Yes | - |
+| `ELASTICSEARCH_API_KEY` | Elasticsearch API key for authentication | Yes | - |
+| `IS_OPENSEARCH` | Flag to indicate OpenSearch instead of Elasticsearch | No | false |
+| `IS_QUICKWIT` | Flag to indicate Quickwit instead of Elasticsearch | No | false |
+| `REDIS_URL` | Redis connection URL | No | - |
+| `REDIS_CLUSTER_ENDPOINTS` | Redis cluster endpoints (comma-separated) | No | - |
+| **S3 Storage** |
+| `USE_S3_STORAGE` | Enable S3-compatible storage for datasets | No | false |
+| `S3_ENDPOINT` | S3-compatible storage endpoint | No | - |
```

Comment:
probably we want to make the S3 endpoint mandatory soon, otherwise they will miss a lot of feature like storing multimodal data, and probably we want to deprecate storing datasets on postgres so we don't need to maintain it forever